packages/babel-preset-env/src/normalize-options.js

```diff
@@ -15,38 +15,51 @@ const validIncludesAndExcludes = new Set([
   ...defaultWebIncludes,
 ]);
 
-export const validateIncludesAndExcludes = (
-  opts: Array<string> = [],
+const pluginToRegExp = (plugin: any): RegExp => {
+  if (plugin instanceof RegExp) return plugin;
+  try {
+    return new RegExp(`^${normalizePluginName(plugin)}$`);
+  } catch (e) {
+    return null;
+  }
+};
+
+const selectPlugins = (regexp: RegExp): Array<string> =>
+  Array.from(validIncludesAndExcludes).filter(
+    item => regexp instanceof RegExp && item.match(regexp),
+  );
+
+const expandIncludesAndExcludes = (
+  plugins: Array<string | RegExp> = [],
   type: string,
 ): Array<string> => {
-  invariant(
-    Array.isArray(opts),
-    `Invalid Option: The '${type}' option must be an Array<String> of plugins/built-ins`,
-  );
+  if (plugins.length === 0) return plugins;
 
-  const unknownOpts = opts.filter(opt => !validIncludesAndExcludes.has(opt));
+  const selectedPlugins = plugins.map(plugin =>
+    selectPlugins(pluginToRegExp(plugin)),
+  );
+  const invalidRegExpList = plugins.filter(
+    (p, i) => selectedPlugins[i].length === 0,
+  );
 
   invariant(
-    unknownOpts.length === 0,
-    `Invalid Option: The plugins/built-ins '${unknownOpts.join(
+    invalidRegExpList.length === 0,
+    `Invalid Option: The plugins/built-ins '${invalidRegExpList.join(
       ", ",
     )}' passed to the '${type}' option are not
     valid. Please check data/[plugin-features|built-in-features].js in babel-preset-env`,
   );
 
-  return opts;
+  return [].concat(...selectedPlugins);
```

Comment:
I guess this is suppose to make a `flatten` operation, it would be good to introduce this utility and use it here
```
const flatten = arr => [].concat(...arr)
```

Reply:
Yeah, good idea 👍

Reply:
Do we have any special place for keeping these kinds of utils?

Reply:
not rly (at least i dont think so), I would just make it a local function in this file for now